dags/remote_automated_ampliseq.py

```diff
@@ -0,0 +1,373 @@
+                local_filepath=local_tar_path,
+                remote_filepath=remote_tar_path,
+                operation='put',
+                create_intermediate_dirs=True
+            )
+            sftp_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def extract_remote_directory(tar_info):
+            logging.info(f"Extracting remote directory for study {tar_info['study']['study']}...")
+            study = tar_info["study"]
+            tar_filename = tar_info["tar_filename"]
+            remote_tar_path = os.path.join(REMOTE_PATH, study['study'], tar_filename)
+            command = f'mkdir -p {REMOTE_PATH}/{study["study"]} && tar --strip-components=1 -xzf {remote_tar_path} -C {REMOTE_PATH}/{study["study"]} && rm {remote_tar_path}'
+            ssh_op = CustomSSHOperator(
+                task_id='extract_remote_directory_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def editing_remote_samplesheet(study):
+            logging.info(f"Editing remote samplesheet for study {study['study']}...")
+            command = f"sed -i 's|{BASE_PATH}/|{REMOTE_PATH}/|g' {REMOTE_PATH}/{study['study']}/samplesheet.csv"
+            ssh_op = CustomSSHOperator(
+                task_id='editing_remote_samplesheet_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def submit_nextflow_script(script_info):
+            logging.info(f"Submitting Nextflow script for study {script_info['study']['study']}...")
+            study = script_info["study"]
+            script_name = script_info["script_name"]
+            command = f'bsub -n 64 -M 250000 -R "rusage[mem=250000] span[hosts=1]" sh {REMOTE_PATH}/{study["study"]}/{script_name}'
+            return CustomSSHOperator(
+                task_id='submit_nextflow_script_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command,
+            ).execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def monitor_status_nextflow(study):
+            logging.info(f"Monitoring Nextflow job status for study {study['study']}...")
+            job_sensor = JobStatusSensor(
+                task_id=f"monitor_status_nextflow_{study['study']}",
```

Comment:
hmmm. based on how the class JobStatusSensor was written i was kind of expecting just one task, rather than one per study. id rather it be one per study, but then JobStatusSensor would have to work a bit differently than it appears to..